### PR TITLE
Added a comment section for each article.

### PR DESCRIPTION
**Pull request by SerhiyToroniy**

### CODE REVIEW

File: sports_hub/sports_hub/Controllers/ArticleController.cs [hunk 1]
```diff
@@ -26,16 +30,172 @@ public IActionResult Index()
...
+
+        [HttpPost]
+        public async Task<JsonResult> LikeComment(string UserId, int ArticleId, int CommentId, bool isMainComment, bool IsIncrement)
+        {
+            if (IsIncrement)
+            {
+                if (isMainComment)
+                {
+                    var mainComment = await _applicationContext.MainComments.FindAsync(CommentId);
+                    mainComment.LikesCount += 1;
```
Comment (XashDM) [inline sports_hub/sports_hub/Controllers/ArticleController.cs:78]: You can use ternary operator IsIncrement  ? mainComment + 1 : mainComment - 1 to simplify

File: sports_hub/sports_hub/Controllers/ArticleController.cs [hunk 2]
```diff
@@ -26,16 +30,172 @@ public IActionResult Index()
...
+                }
+            }
+            await _applicationContext.SaveChangesAsync();
+            return Json(true);
+        }
+
+        [HttpPost]
+        public async Task<JsonResult> DislikeComment(string UserId, int ArticleId, int CommentId, bool isMainComment, bool IsIncrement)
+        {
+            if (IsIncrement)
```
Comment (XashDM) [inline sports_hub/sports_hub/Controllers/ArticleController.cs:106]: Same as before

File: sports_hub/sports_hub/Controllers/ArticleController.cs [hunk 3]
```diff
@@ -26,16 +30,172 @@ public IActionResult Index()
...
+
         [HttpGet]
-        public ActionResult GetArticle([FromQuery] int id)
+        public ActionResult GetArticle(int id, int commentsSortedBy, int visisbleCommentsCounter)
         {
             ViewBag.AvailableLanguages = _applicationContext.Languages.ToList();
             ViewBag.urlForShareButtons = Constants.ShareButtonsURL;
+
+            const int minCommentsCount = 3;
+            const int maxCommentsCount = 20;
```
Comment (XashDM) [inline sports_hub/sports_hub/Controllers/ArticleController.cs:143]: Move to constants

File: sports_hub/sports_hub/Models/ApplicationContext.cs [hunk 4]
```diff
@@ -30,9 +34,41 @@ public ApplicationContext(DbContextOptions<ApplicationContext> options)
...
+            //    .Metadata
+            //    .SetValueComparer(valueComparer);
+
+            //modelBuilder.Entity<BaseComment>()
+            //    .Property(e => e.DislikesCount)
+            //    .Metadata
+            //    .SetValueComparer(valueComparer);
+
+            //modelBuilder.Entity<BaseComment>()
+            //    .HasNoKey();
```
Comment (XashDM) [inline sports_hub/sports_hub/Models/ApplicationContext.cs:70]: remove

File: sports_hub/sports_hub/wwwroot/js/ArticleScript.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+﻿const inputField = document.getElementsByClassName("input-box-txt")[0];
+const btn = document.getElementsByClassName("input-box-cancel")[0];
+var firstReplyAppearance = true;
+//let sumHeight = 0;
```
Comment (XashDM) [inline sports_hub/sports_hub/wwwroot/js/ArticleScript.js:4]: remove commented lines

**Review by XashDM — APPROVED** [review]
Fix comments, you can merge after that

File: sports_hub/sports_hub/Models/ApplicationContext.cs (hunk 4 above)
Comment (SerhiyToroniy) [reply sports_hub/sports_hub/Models/ApplicationContext.cs:70]: I left this code for further implementations.

**Pull request merged**